### PR TITLE
Add java floor

### LINKED ISSUE #899: [Term Entry] Java Math Methods: `.floor()`
### Reason/inspiration (optional)

Under `java/concepts/math-methods/terms/`, create a new term entry for the `Math.floor()` Java method.

- In the `terms/` folder, make a new folder called `floor`.
- Inside the `floor` folder, create a new markdown file called `floor.md`
- Inside the file, be sure to include a "Syntax" section with a \```pseudo code snippet.
- Also make sure to include an "Example" section with a \```java code snippet.

Please refer to the [term entry template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by J0hnRJr**
### Description

New .md file to describe and explain the Math.floor() function in Java. Closes #899 

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR.

### CODE REVIEW

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [inline content/java/concepts/math-methods/terms/floor/floor.md:16]:
```suggestion
## Syntax

The Math package can be imported and utilized as shown below

```

For headers and code blocks, let's make sure we have an empty newline above _and_ below it. Can we make this edit throughout the rest of this file?

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [inline content/java/concepts/math-methods/terms/floor/floor.md:13]:
```suggestion
The **`.floor()`** method returns the largest integer value that is less than or equal to the argument.
```

I think the information in the second sentence fits better in the "Syntax" section.

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [inline content/java/concepts/math-methods/terms/floor/floor.md:17]:
```suggestion
```pseudo
```

Let's use \```pseudo code blocks for the "Syntax" section. 😄

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [inline content/java/concepts/math-methods/terms/floor/floor.md:22]:
```suggestion
Math.floor(x)
```

For the code block in the "Syntax" section, let's try to keep it to 1-2 lines, in most cases. This should be a place to display the general structure of the concept/term being presented.

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [inline content/java/concepts/math-methods/terms/floor/floor.md:24]:
```suggestion

The return value is the same as `num` under the following conditions:

- `num` is already an integer.
- `num` is `NaN`, infinity, or zero (positive or negative).

```

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [inline content/java/concepts/math-methods/terms/floor/floor.md:25]: Let's consolidate these into two examples that fit this general structure/format:

## Example 1

[Sentence(s)]

[\```java code block] <--- Successful calls that include negative values.

## Example 2

[Sentence(s)]

[\```java code block] <-- Unsuccessful calls that include `NaN` and Infinity.

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [inline content/java/concepts/math-methods/terms/floor/floor.md:10]:
```suggestion
  - 'learn-java'
  -  'paths/computer-science'
```

Just the parent directory of the free course slug is sufficient. Let's also include a paid path/course slug. 😄

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (J0hnRJr) [reply content/java/concepts/math-methods/terms/floor/floor.md:23]: Integrated suggested change.

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (J0hnRJr) [reply content/java/concepts/math-methods/terms/floor/floor.md:29]: Simplified and consolidated examples as suggested.

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (J0hnRJr) [reply content/java/concepts/math-methods/terms/floor/floor.md]: Implemented suggested change.

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (J0hnRJr) [reply content/java/concepts/math-methods/terms/floor/floor.md]: Suggestion implemented.

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (J0hnRJr) [reply content/java/concepts/math-methods/terms/floor/floor.md]: Made the change as requested.

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (J0hnRJr) [reply content/java/concepts/math-methods/terms/floor/floor.md]: Yes, I spaced out the code blocks and headers.

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (J0hnRJr) [reply content/java/concepts/math-methods/terms/floor/floor.md]: This is moved to the Syntax section now.

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (HishamT) [inline content/java/concepts/math-methods/terms/floor/floor.md:14]:
```suggestion
The **`Math.floor()`** method included in the `Math` class returns the largest integer value that is less than or equal to the argument.
```

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (HishamT) [inline content/java/concepts/math-methods/terms/floor/floor.md:18]: I'm not sure what you are trying to say here.

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (HishamT) [inline content/java/concepts/math-methods/terms/floor/floor.md:31]:
```suggestion
In the case of double values, `Math.floor()` returns the next integer value below the provided double value. In the case of integers provided, these will return the same value as provided:
```

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (HishamT) [inline content/java/concepts/math-methods/terms/floor/floor.md:45]: This doesn't return infinity. This causes an `ArithmeticException` to be thrown.

**Review by HishamT — CHANGES_REQUESTED** [review]
I have finished the second review of this entry. Comments are left for you.

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (J0hnRJr) [reply content/java/concepts/math-methods/terms/floor/floor.md]: I removed this line as it was already stated further down in the Syntax section.

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (J0hnRJr) [reply content/java/concepts/math-methods/terms/floor/floor.md]: I changed the provided example value to one I tested which produces the result Infinity.

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [inline content/java/concepts/math-methods/terms/floor/floor.md:8]:
```suggestion
  - 'Documentation'
  - 'Numbers'
```

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [inline content/java/concepts/math-methods/terms/floor/floor.md:18]:
```suggestion
## Syntax

```

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [inline content/java/concepts/math-methods/terms/floor/floor.md:20]:
```suggestion
Math.floor(x);
```

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [inline content/java/concepts/math-methods/terms/floor/floor.md:23]:
```suggestion
The `num` parameter can either be a floating-point value or a variable that points to a floating-point value. The return value is the same as `num` under the following conditions:
```

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [inline content/java/concepts/math-methods/terms/floor/floor.md:32]:
```suggestion
```java
```

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [inline content/java/concepts/math-methods/terms/floor/floor.md:14]:
```suggestion
The **`Math.floor()`** method returns the largest integer value that is less than or equal to the argument.
```

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [inline content/java/concepts/math-methods/terms/floor/floor.md:35]:
```suggestion
// Example.java
public class Example {
  public static void main(String args[]) {
    System.out.println(Math.floor(-3.9));
    System.out.println(Math.floor(1.0001));
    System.out.println(Math.floor(0));
  }
}
```

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [inline content/java/concepts/math-methods/terms/floor/floor.md:37]:
```suggestion

This will produce the following output:

```

And then let's add a \```shell snippet that looks like this:

```shell
-4.0
1.0
0.0
```

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [inline content/java/concepts/math-methods/terms/floor/floor.md:40]:
```suggestion
The following example features unsuccessful calls to the `.floor()` method:
```

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [inline content/java/concepts/math-methods/terms/floor/floor.md:42]:
```suggestion
```pseudo
```

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [inline content/java/concepts/math-methods/terms/floor/floor.md:44]:
```suggestion
public class Example {
  public static void main(String args[]) {
    System.out.println(Math.floor(2.0 % 0));
    System.out.println(Math.floor(Double.POSITIVE_INFINITY));
  }
}
```

And then underneath let's add the following sentence:

> This will produce the following output:

And then a \```shell snippet that looks like this:

```shell
NaN
Infinity
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@J0hnRJr left a few last comments before we merge. 

@HishamT anything left to add, change, or check?

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (J0hnRJr) [reply content/java/concepts/math-methods/terms/floor/floor.md:31]: @Dusch4593 in the first review you asked for these to be pseudo:
> Let's use ```pseudo code blocks for the "Syntax" section. 😄
Just want to double check before reversing the previous change.

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (J0hnRJr) [reply content/java/concepts/math-methods/terms/floor/floor.md]: Implemented in commit [b59b2e74](https://github.com/Codecademy/docs/pull/1174/commits/b59b2e741d9151889488652b2133242ccf23f1f2)

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (J0hnRJr) [reply content/java/concepts/math-methods/terms/floor/floor.md]: used `Math.floor()` in place of `.floor()` and added the suggestion to commit [db80c90e](https://github.com/Codecademy/docs/pull/1174/commits/db80c90ea1286c0668b1bea74b1901fd8f74437e)

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [reply content/java/concepts/math-methods/terms/floor/floor.md:55]: @J0hnRJr Haha yes, my mistake. 😂 Let's change to \```java

File: content/java/concepts/math-methods/terms/floor/floor.md
Comment (Dusch4593) [reply content/java/concepts/math-methods/terms/floor/floor.md:32]: @J0hnRJr This code block is not in the "Syntax" section; it is in the "Example 1" section and it should feature a \```java code block.

\```pseudo code blocks are strictly meant for describing the syntax of something as opposed to showcasing example of that something being applied.

**Review by Dusch4593 — APPROVED** [review]
@J0hnRJr With the exception of [this suggestion](https://github.com/Codecademy/docs/pull/1174/files#r1009318000) and [that suggestion](https://github.com/Codecademy/docs/pull/1174/files#r1009499060), everything LGTM! 👍🏻

@HishamT, is there anything else to add, change, or check?

**Comment by J0hnRJr** [comment]
> @J0hnRJr With the exception of [this suggestion](https://github.com/Codecademy/docs/pull/1174/files#r1009318000) and [that suggestion](https://github.com/Codecademy/docs/pull/1174/files#r1009499060), everything LGTM! 👍🏻
> 
> @HishamT, is there anything else to add, change, or check?

Should be addressed now. Ready to go!

**Comment by Dusch4593** [comment]
Congratulations and thank you so much for your contribution to Codecademy Docs, @J0hnRJr! 🚀 🥳 🎉

The entry should now be live: https://www.codecademy.com/resources/docs/java/math-methods/floor

Screen Shot 2022-10-31 at 2 11 30 PM

Looking forward to your next contribution to Docs! 😄

**Comment by Dusch4593** [comment]
Hey @J0hnRJr! 👋🏻

I was wondering if you'd be open to creating a new PR to make some edits to the [`.floor()` entry you just contributed](https://www.codecademy.com/resources/docs/java/math-methods/floor)?

- On line # 2, can we change the `Title` to `.floor()`?

**Pull request merged**
